koku/masu/database/report_db_accessor_base.py

```diff
@@ -133,21 +132,33 @@ def _execute_trino_raw_sql_query_with_description(
             ctx = {}
 
         sql, bind_params = self.trino_prepare_query(sql, sql_params)
-        t1 = time.time()
-        trino_conn = trino_db.connect(schema=self.schema, **conn_params)
-        LOG.info(log_json(msg="executing trino sql", log_ref=log_ref, context=ctx))
-        try:
-            trino_cur = trino_conn.cursor()
-            trino_cur.execute(sql, bind_params)
-            results = trino_cur.fetchall()
-            description = trino_cur.description
-        except Exception as ex:
-            if attempts_left == 0:
-                LOG.error(log_json(msg="failed trino sql execution", log_ref=log_ref, context=ctx), exc_info=ex)
-            raise ex
-        running_time = time.time() - t1
-        LOG.info(log_json(msg="executed trino sql", log_ref=log_ref, running_time=running_time, context=ctx))
-        return results, description
+        attempts_left = max(attempts_left, 1)
+
+        for i in range(attempts_left):
+            t1 = time.time()
+            trino_conn = trino_db.connect(schema=self.schema, **conn_params)
+            LOG.info(log_json(msg="executing trino sql", log_ref=log_ref, context=ctx))
+            try:
+                trino_cur = trino_conn.cursor()
+                trino_cur.execute(sql, bind_params)
+                results = trino_cur.fetchall()
+                description = trino_cur.description
+                running_time = time.time() - t1
+                LOG.info(log_json(msg="executed trino sql", log_ref=log_ref, running_time=running_time, context=ctx))
+                return results, description
+            except Exception as ex:
+                if attempts_left > 0:
```

Comment:
Are you sure this works as expected? I could be wrong but I don't see any change to the original attempts_left value so we will never get to 0 and stop retrying? I think your missing something like `attempts_left=(attempts_left - 1) - i` somewhere.
```
attempts = 3
for i in range(attempts):
    if attempts > 0:
        print(True)
    print(i)

Result:
True
0
True
1
True
2

Reply:
It wasn't correct. I've fixed it! Thanks.